tls/s2n_ktls.h

```diff
@@ -17,19 +17,32 @@
 
 #include "tls/s2n_config.h"
 
+/* Define headers needed to enable and use kTLS.
+ *
+ * The inline header definitions are required to compile kTLS specific code.
+ * kTLS has been tested on linux. For all other platforms, kTLS is marked as
+ * unsupported, and will return an unsupported error.
+ */
+#if defined(__linux__)
+    #define S2N_KTLS_SUPPORTED true
+    #include "tls/s2n_ktls_linux.h"
+#else
+    #define S2N_KTLS_SUPPORTED false
+    #include "tls/s2n_ktls_unsupported.h"
+#endif
+
 /* A set of kTLS configurations representing the combination of sending
  * and receiving.
  */
 typedef enum {
-    /* Disable kTLS. */
-    S2N_KTLS_MODE_DISABLED,
     /* Enable kTLS for the send socket. */
     S2N_KTLS_MODE_SEND,
     /* Enable kTLS for the receive socket. */
     S2N_KTLS_MODE_RECV,
-    /* Enable kTLS for both receive and send sockets. */
-    S2N_KTLS_MODE_DUPLEX,
 } s2n_ktls_mode;
 
-int s2n_config_set_ktls_mode(struct s2n_config *config, s2n_ktls_mode ktls_mode);
 bool s2n_ktls_is_supported_on_platform();
+
+/* These functions will be moved to api/s2n.h when the kTLS feature is released. */
```

Comment:
No plans to release them as unstable?

Reply:
I wasnt differentiating in the comment but yes its probably unstable for a small duration. Comment was more to indicate whats going to be public for reviewers.